flags.ts

```diff
@@ -1,24 +1,80 @@
-import { Flag } from "$live/types.ts";
+import { MiddlewareHandlerContext } from "$fresh/server.ts";
+import { DecoManifest, Flag, LiveState } from "$live/types.ts";
 import { context } from "$live/live.ts";
-import getSupabaseClient from "./supabase.ts";
+import getSupabaseClient from "$live/supabase.ts";
+import { EffectFunction, MatchFunction } from "$live/std/types.ts";
+import RandomMatch from "$live/functions/RandomMatch.ts";
+import SelectPageEffect from "$live/functions/SelectPageEffect.ts";
 
 let flags: Flag[];
 export const flag = (id: string) => flags.find((flag) => flag.id === id);
 
-export const ensureFlags = async () => {
-  const site = context.site;
+export const loadFlags = async (
+  req: Request,
+  ctx: MiddlewareHandlerContext<LiveState>,
+) => {
+  const site = context.siteId;
+  const manifest = context.manifest as DecoManifest;
 
   // TODO: Cache flags stale for 5 minutes, refresh every 30s
-  const { data: Flags, error } = await getSupabaseClient()
-    .from("flags")
+  const { data: availableFlags, error } = await getSupabaseClient()
+    .from<Flag>("flags")
     .select(
-      `id, name, audience, traffic, site!inner(name, id), pages!inner(data, path, id)`,
+      `id, name, key, state, site!inner(name, id), data`,
```

Comment:
Aren't site's `name` and `id` already available at the Live Context?

I think we can avoid a DB join here.